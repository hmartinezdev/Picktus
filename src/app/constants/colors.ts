export default {
  black: '#151614',
  blue: '#9fcbe7',
  error: '#982649',
  green: '#75ab88',
  lightGreen: '#b3e099',
  primary: '#2d3047',
  secondary: '#FAC9B8',
  warning: '#E37E5B',
  white: '#F8F8F7',
};
