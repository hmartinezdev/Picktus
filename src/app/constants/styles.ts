export const borderRadius = '3px';
export const fontFamily = "'Josefin Sans', sans-serif";
export const boxShadow = '11px 10px 34px -7px rgba(0, 0, 0, 0.49)';
