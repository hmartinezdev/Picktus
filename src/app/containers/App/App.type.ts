export interface IReduxStateProps {
  showLoader: boolean;
}
