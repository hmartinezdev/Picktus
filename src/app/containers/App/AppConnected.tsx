import { IReduxState } from '@store/reducers';
import { userLoginSuccess } from '@store/reducers/user/user-actions';
import { connect } from 'react-redux';
import App from './App';
import { IReduxStateProps } from './App.type';

const mapStateToProps = (state: IReduxState): IReduxStateProps => {
  return {
    showLoader: state.display.loader,
  };
};

export default connect(
  mapStateToProps,
  { userLoginSuccess }
)(App);
