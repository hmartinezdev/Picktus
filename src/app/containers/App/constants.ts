export const config = {
  apiKey: 'AIzaSyAVQ-Abn_Sf0z18jPeW_mKhLOwwGfWn_40',
  authDomain: 'picktus-76101.firebaseapp.com',
  databaseURL: 'https://picktus-76101.firebaseio.com',
  projectId: 'picktus-76101',
  storageBucket: 'picktus-76101.appspot.com',
  messagingSenderId: '1077268436650',
};
