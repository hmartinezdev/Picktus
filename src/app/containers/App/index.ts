import { IReduxState } from '@store/reducers';
import { connect } from 'react-redux';
import App from './App';
import { IReduxStateProps } from './App.type';

const mapStateToProps = (state: IReduxState): IReduxStateProps => {
  return {
    showLoader: state.display.loader,
  };
};

export default connect(mapStateToProps)(App);
