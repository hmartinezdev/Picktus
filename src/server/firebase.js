/* tslint:disable: max-line-length */

module.exports = {
  auth_uri: 'https://accounts.google.com/o/oauth2/auth',
  client_email: 'firebase-adminsdk-vkv42@picktus-76101.iam.gserviceaccount.com',
  client_id: '106852938874480059583',
  client_x509_cert_url:
    'https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-vkv42%40picktus-76101.iam.gserviceaccount.com',
  private_key:
    '-----BEGIN PRIVATE KEY-----\nMIIEvwIBADANBgkqhkiG9w0BAQEFAASCBKkwggSlAgEAAoIBAQDQ7AF17FSf7fac\n2tHFfrmMr/U+tmNhjjT9RUlEXcq+hExx+5aqPqP4RZvWxCqPrkRfxLUfRgmDnY4l\nR37JDEU8uA4jvvT/CRvGP65Q7oWVSFEvGinXAV+2JcYxZxuvypfb+f8vVz1ZJvjV\n9tCPNuPvr65tVnz6rR9G8bdoxLOmO4sWdUBt5HBR1P960hLVyMLF2TI0qV21Zfax\n6/J5JQh4cxnAVH9alFVYTwGGyaMK54J3nICrqC4tIL9orIjr4f0Ree49Koa2nK1S\nqrFL3nKlYuQ49MT5S8bsEkOZDI89x0cbE72z+9KSYX0wZisuqQS/mnwLgvo/V5KV\nyhmlaqp7AgMBAAECggEAGq0HwrbaV1FbG8wU3NLBfmU5Oziul9ZAjdY69K/FI25h\nFGy7YCF/bkP1EgcaKB5oPJIxTjzzGgsUC06FscYPUSFvLMFFJwtSsI2RBGRwSg9h\nzlHY5csXHXrVQ5mWv2tu2HejTbgl7Sw8uZU6LA8mK4STEQLqElfolnZmpMn45lCY\n3OejiCHlFiARiRV56Jr7DO5AlZzklusBXitDHUhIzCLBjjH0KdgqRrByRfSMNRTV\nawggV9ysqI5iOIAKdJXNTr6CoOBAAl7YJLiJ71y0KsNUOU06/Z6eCNAqcqS1yl/e\nCIzdsk3bnoZ8GG/TxjevYjf4/PfdXy+Q+QNFaWzwAQKBgQD0Di/ka1+5rVGNKeGU\noS6+EpFVXazrGFwxA6QQV/VXxl1aLIMqLJDmSr+TTHEJbYwV7WXIrcF+aewsgDkh\nUH2OJ7t36g4n1uEV0ZhzegymrGWwHlZFFVNp4Hmez2hbz+ibQTdPGBA+F8qNR8KK\neNkreEuShXE+ehqOrZZghN5s+wKBgQDbJZ/mim3K9KGPZ7sMNuZEF1cc2ArR8jiC\nV0F7E58bRJOdxiCOk/A3xr8gNEN8cMhB9Xxd7VmTs7GNUCb0b9IFC6V03/58Ka4N\n2aWeAZX4xtUYy8SMSP/Wsw4FIUjmHmZ9lP/5KIm/LLKDRr6qUKCv2+LiB8GhnLRx\nx0bceJVAgQKBgQDstFHY/hcy1BnzivpCGG5oZUPWAuvyfIb19u0J+PV482iZS/DP\nV7gV08upBovftAMZRU3h7Dxt6jfpFVfTquiLlpepfzXJiAx5B/cPkQFyKmTaAyJz\nEHL9jn0G/WzfF3AoHIGTUeh3AMYwbrgNa5ZeJedOLeSezvixB0WBSX0dPwKBgQCw\nTwbkPX6ziDjq3POZkhmTsot9QzsBSbS4+GZQFKsvDu8MQjgzPUZIOXfO+v1Igpw6\npLvX/XrR4AH6xP9yLPx03rfQ3d0qG4ZS0kXyZyg3LBCJkHaivqBGJWwkKHEz4kVm\nx5dg54Ja8iT0hNgkwhVlJqzUWJUp5p8t4rtZ6RF7AQKBgQC+7c90RFR7CEuJp8vQ\njjn8n+xGuaHXXEMEgarCGvLOs8UUTa5OrRDXY91KIyxh7hdUYLEMiR0XP4wIDF91\nflRfCE2SXsW33e/jd+oZkYCg5w1Oofj4XAyp0QoRRE60Z0MCPceq86tUVy0N5oeD\nyXokhCB/hpT3XtH2GXZiqhwQ3g==\n-----END PRIVATE KEY-----\n',
  private_key_id: '3140acfc3ecc44fc2114f9d17512fd82d49d1ac7',
  project_id: 'picktus-76101',

  token_uri: 'https://oauth2.googleapis.com/token',
  type: 'service_account',
};
