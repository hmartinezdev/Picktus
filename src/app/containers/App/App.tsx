import Background from '@components/Background';
import Header from '@components/Header';
import Loader from '@components/Loader';
import Messages from '@components/Messages';
import firebase from 'firebase/app';
import { withRouter } from 'next/router';
import React, { Component } from 'react';
import { Transition, TransitionGroup } from 'react-transition-group';
import { IAppPropsType } from './App.type';
import { config } from './constants';

class App extends Component<IAppPropsType> {
  constructor(props: IAppPropsType) {
    super(props);

    // If the firebase setup has not been done yet
    // we do it here
    if (!firebase.apps.length) {
      firebase.initializeApp(config);
    }
  }

  public render(): React.ReactElement<App> {
    const { children, showLoader, router } = this.props;

    return (
      <Background>
        <div className="page">
          <Header />
          <Messages />
          {showLoader && <Loader />}
          <TransitionGroup className="app_transitiongroup" appear={false}>
            {React.Children.map(children, (child) => (
              <Transition key={router.route} timeout={600} mountOnEnter={true} unmountOnExit={true} appear={false}>
                {(status) => React.cloneElement(child as React.ReactElement<any>, { status })}
              </Transition>
            ))}
          </TransitionGroup>
          <style jsx>{`
            .page {
              overflow: hidden;
              min-height: 100vh;
              display: flex;
              align-self: stretch;
              overflow: hidden;
              width: 100%;
            }

            :global(.app_transitiongroup) {
              overflow: hidden;
              min-height: 100vh;
              display: flex;
              align-self: stretch;
              overflow: hidden;
              width: 100%;
            }

            :global(*) {
              padding: 0;
              margin: 0;
            }

            :global(html) {
              font-size: 100%;
            }
          `}</style>
        </div>
      </Background>
    );
  }
}

export default withRouter(App);
