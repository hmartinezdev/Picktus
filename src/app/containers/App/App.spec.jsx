import { mount, shallow } from 'enzyme';
import firebase, { initializeApp } from 'firebase/app';
import React from 'react';
import App from './App';
import { config } from './constants';

let wrapper;
let spy;

const setup = () =>
  shallow(
    <App>
      <div />
    </App>
  );
describe('<App />', () => {
  beforeAll(() => {
    spy = jest.spyOn(firebase, 'initializeApp').mockImplementation(() => {
      return;
    });
    wrapper = setup();
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  test('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    beforeEach(() => {
      spy = jest.spyOn(firebase, 'initializeApp').mockImplementation(() => {
        return;
      });
    });

    test('it should configure firebase properly', () => {
      wrapper = setup();
      expect(spy).toHaveBeenCalledWith(config);
    });

    test('it should configure firebase properly', () => {
      const switcher = firebase;
      firebase = { initializeApp: () => 'test', apps: ['test'] };
      wrapper = setup();
      expect(spy).toHaveBeenCalledTimes(0);
      firebase = switcher;
    });
  });
});
