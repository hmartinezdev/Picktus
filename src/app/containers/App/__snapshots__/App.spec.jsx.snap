// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render loader is showLoader is at true 1`] = `
<div
  className="page"
>
  <Header />
  <Loader />
  <div />
  <style
    jsx={true}
  >
    
          .page {
            min-height: 100vh;
            background-color: #2d3047;
          }
          :global(*) {
            padding: 0;
            margin: 0;
          }

          :global(html) {
            font-size: 100%;
          }
        
  </style>
</div>
`;

exports[`<App /> should render properly 1`] = `
<div
  className="page"
>
  <Header />
  <div />
  <style
    jsx={true}
  >
    
          .page {
            min-height: 100vh;
            background-color: #2d3047;
          }
          :global(*) {
            padding: 0;
            margin: 0;
          }

          :global(html) {
            font-size: 100%;
          }
        
  </style>
</div>
`;
